### PR TITLE
fix(tax): Fix tax blue/green

### PR DESCRIPTION
**Pull request by flooie**
Fix Blue Green Deploy issue with tax court

### CODE REVIEW

File: juriscraper/opinions/united_states/federal_special/tax.py [hunk 1]
```diff
@@ -30,6 +31,22 @@ def __init__(self, *args, **kwargs):
         }
 
     def _download(self, request_dict={}):
+        """Download from api
+
+        The tax court switches between blue and green deploys so we need to
+        check which one is current before we continue
+
+        :param request_dict:
+        :return: None
```
Comment (mlissner) [inline juriscraper/opinions/united_states/federal_special/tax.py:40]: Don't forget these, please.

File: juriscraper/opinions/united_states/federal_special/tax.py [hunk 2]
```diff
@@ -30,6 +31,22 @@ def __init__(self, *args, **kwargs):
...
+        """
+        if not self.set_blue_green and not self.test_mode_enabled():
+            check = self.request["session"].get(self.url)
+            if check.status_code != 200:
```
Comment (mlissner) [inline juriscraper/opinions/united_states/federal_special/tax.py:44]: Every get request must have a timeout. If we forget these, the code can hang, literally forever, which would stop our scraper for all courts. It's a massive wart in the requests library that they're incapable of fixing.

**Review by mlissner — COMMENTED** [review]
I was just coming to see the change and gloat about green/blue biting us as I thought it might, but I did see something that's important to fix, please.

File: juriscraper/opinions/united_states/federal_special/tax.py (hunk 1 above)
Comment (flooie) [reply juriscraper/opinions/united_states/federal_special/tax.py:40]: yes.

File: juriscraper/opinions/united_states/federal_special/tax.py (hunk 2 above)
Comment (flooie) [reply juriscraper/opinions/united_states/federal_special/tax.py:44]: I assume it's subclassed with a timeout.

Comment (mlissner) [reply juriscraper/opinions/united_states/federal_special/tax.py:44]: Oh, maybe you're right. I'm pretty paranoid about it.

Comment (flooie) [reply juriscraper/opinions/united_states/federal_special/tax.py:44]:
```
    def _request_url_get(self, url):
        """Execute GET request and assign appropriate request dictionary
        values
        """
        self.request["url"] = url
        self.request["response"] = self.request["session"].get(
            url,
            headers=self.request["headers"],
            verify=self.request["verify"],
            timeout=60,
            **self.request["parameters"],
        )

    def _request_url_post(self, url):
        """Execute POST request and assign appropriate request dictionary values"""
        self.request["url"] = url
        self.request["response"] = self.request["session"].post(
            url,
            headers=self.request["headers"],
            verify=self.request["verify"],
            data=self.parameters,
            timeout=60,
            **self.request["parameters"],
        )
```

Comment (flooie) [reply juriscraper/opinions/united_states/federal_special/tax.py:44]: Just to square the circle, I believe you were already paranoid enough about timeouts within juriscraper

File: juriscraper/opinions/united_states/federal_special/tax.py (hunk 1 above)
Comment (flooie) [reply juriscraper/opinions/united_states/federal_special/tax.py:40]: Sounds good.  I'll probably just delete the param completely as I don't think it's needed .

**Comment by flooie** [comment]
And yes - I knew you'd be pleased about the green/blue thing.  But ... we now have a fix for it - (I hope).

**Comment by mlissner** [comment]
It's always nice to see the world working like you think it does. Also, hey, they did a deployment at a court!

**Comment by flooie** [comment]
Although, it would be nice if they would ever respond to any of our GitHub issues

**Pull request merged**